packages/base-driver/lib/protocol/protocol.js

```diff
@@ -42,14 +42,16 @@ function isSessionCommand (command) {
 }
 
 function getLogger (driver, sessionId = null) {
-  if (_.isFunction(driver?.log?.info)) {
-    return driver.log;
+  const dstDriver = sessionId && _.isFunction(driver.driverForSession)
+    ? (driver.driverForSession(sessionId) ?? driver)
+    : driver;
+  if (_.isFunction(dstDriver.log?.info)) {
+    return dstDriver.log;
   }
 
-  let logPrefix = 'AppiumDriver';
-  if (driver && driver.constructor) {
-    logPrefix = driver.constructor.name;
-  }
+  let logPrefix = dstDriver.constructor
+    ? `${dstDriver.constructor.name}@${node.getObjectId(dstDriver).substring(0, 8)}`
+    : 'AppiumDriver';
```

Comment:
or `BaseDriver` as this package name? (maybe `Appium` prefix might be redundant)

Reply:
I use this one, because it is aligned with other loggers in this module. Anyway, I don't think there ever might be a situation where the destination driver does not have a constructor